hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/format/mor/MergeOnReadInputFormat.java

```diff
@@ -298,8 +299,14 @@ private ParquetColumnarRowSplitReader getReader(String path, int[] requiredPos)
         new org.apache.hadoop.fs.Path(path).getParent(),
         this.conf.getBoolean(FlinkOptions.HIVE_STYLE_PARTITIONING),
         FilePathUtils.extractPartitionKeys(this.conf));
-    LinkedHashMap<String, Object> partObjects = new LinkedHashMap<>();
-    partSpec.forEach((k, v) -> partObjects.put(k, DataTypeUtils.resolvePartition(
+
```

Comment:
Thanks, why not fix the partObjects directly when generating it?

Reply:
Looks a good way! Modified.